### PR TITLE
mod-spring-template readme file.

### PR DESCRIPTION
**Pull request by TarasSpashchenko**
Added how-to part to the readme file to be used as a guideline for those who want to create a new spring-based module in folio.

### CODE REVIEW

**Review by dcrossleyau — CHANGES_REQUESTED** [review]
Being a template, this should ideally have Java 17 as the base rather than old Java 11:
https://dev.folio.org/faqs/how-to-specify-backend-java-ci/

Should then also use 'wget' for the "health check" rather than 'curl' [FOLIO-3507](https://issues.folio.org/browse/FOLIO-3507).

File: README.md [hunk 1]
```diff
@@ -6,3 +6,70 @@ This software is distributed under the terms of the Apache License,
...
+## Introduction
+
+This is the blueprint module that should be used as a template for FOLIO Spring-based backend modules.
+It provides the initial configuration and common data schemas used by the FOLIO platform.
+
+
+## How-to guide
+This guide shows how to create a new folio backend module using the mod-spring-template engine
+
+1. Clone the mod-sping-template repository from the github https://github.com/folio-org/mod-spring-template
```
Comment (julianladisch) [inline README.md]:
```suggestion
1. Clone the mod-spring-template repository from the github https://github.com/folio-org/mod-spring-template
```

File: README.md [hunk 2]
```diff
@@ -6,3 +6,70 @@ This software is distributed under the terms of the Apache License,
...
+
+- Spring Framework https://spring.io/projects/spring-framework that includes:
+  - Core technologies: dependency injection, events, resources, i18n, validation, data binding, type conversion, SpEL, AOP.
+  - Testing: mock objects, TestContext framework, Spring MVC Test, WebTestClient.
+  - Data Access: transactions, DAO support, JDBC, ORM, Marshalling XML.
+  - Spring MVC and Spring WebFlux web frameworks.
+  - Integration: remoting, JMS, JCA, JMX, email, tasks, scheduling, cache.
+- Spring Boot https://spring.io/projects/spring-boot
+- Spring Data JPA https://spring.io/projects/spring-data-jpa
+- Spring for Apache Kafka https://spring.io/projects/spring-kafka
```
Comment (psmagin) [inline README.md:74]: Spring Batch also

File: README.md (hunk 1 above)
Comment (TarasSpashchenko) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (TarasSpashchenko) [reply README.md:74]: Done

**Review by dcrossleyau — APPROVED** [review]

**Review by dcrossleyau — COMMENTED** [review]
The top of the README is not correct and will lead to bad practice.

As explained https://dev.folio.org/guidelines/create-new-repo/ and [FOLIO-360](https://issues.folio.org/browse/FOLIO-360): Use the initial year of creation for the date. (In subsequent years it will become a range.)

At mod-rmb-template, a maven archetype is used for initialising a new module. It has the single date of creation.

**Pull request merged**